app/controllers/users.js

```diff
@@ -1,66 +1,43 @@
 const bcrypt = require('bcryptjs');
 
 const { User } = require('../models');
-const errors = require('../middlewares/errors');
-const logger = require('../logger');
+const errors = require('../errors');
 
-// Verifica la validez de la contraseña (alfanumérica > 7)
 const passwordLength = password => {
   const characters = /^[a-z0-9]+$/i;
-  const alfanumeric = characters.test(password);
-  const isValid = password.length > 7 && alfanumeric;
-  return isValid;
+  return password.length > 7 && characters.test(password);
 };
 
-// Hashea el password mediante bcrypt
 const hashPassword = async password => {
   const salt = await bcrypt.genSaltSync(10);
   const hashedPass = await bcrypt.hashSync(password, salt);
   return hashedPass;
 };
 
-// Verifica la validez y el dominio del email
-const emailValidation = email => {
-  const stringLength = email.length;
-  const testLength = stringLength > 13;
-  const testDomain = email.substring(stringLength - 13) === '@wolox.com.ar';
-  const isValid = testLength && testDomain;
-  return isValid;
-};
-
-const mapRequest = body => {
-  const lastName = body.last_name;
-  return lastName;
-};
+const emailValidation = email => email.length > 13 && email.substring(email.length - 13) === '@wolox.com.ar';
 
 exports.register = async (req, res, next) => {
-  const { email, password, name } = req.body;
-  const lastName = mapRequest(req.body);
+  const { email: dirtEmail, password, name, last_name: lastName } = req.body;
+  const email = dirtEmail.trim();
   try {
-    // Verifico que el request contenga todos los campos
     const userValid = email && password && name && lastName;
-    if (!userValid) res.json({ error: 'Missing fields.' });
+    if (!userValid) throw errors.createUserError('Missing fields.');
 
-    // Verifico que el mail sea válido
     const validEmail = emailValidation(email);
-    if (!validEmail) res.json({ error: 'Invalid email.' });
+    if (!validEmail) throw errors.createUserError('Invalid email.');
+
+    const validPass = passwordLength(password);
+    if (!validPass) throw errors.createUserError('Invalid password.');
 
-    // Verifico que el email no exista previamente
     const emailExists = await User.findOne({ where: { email } });
-    if (emailExists) return res.send({ error: 'Email already exists.' });
+    if (emailExists) throw errors.createUserError('Email already exists.');
```

Comment:
In this case, the email exists... so here could be uniqueEmailError. Are you agree?